docs/quickstart.md

```diff
@@ -0,0 +1,141 @@
+# quickstart
+As described in the [README](https://github.com/neume-network/strategies/readme.md), a strategy requires an `extractor` and, optionally, a `transformer`:
+
+- every run of the `update` function of the `extractor` should pull a chunk of the data from the source, be it a page, a time range etc
+- once the extraction process is finished, the transformation part begins. This is executed on a per-line basis (`onLine`) and should prepare the data to be compliant with the schema of the final dataset.
+
+Depending on the data source, different initial configurations may be needed.
+
+## https strategy
+A web 2 strategy crawls data from an `https` source (rss feed, apis, etc...). 
+
+This guide will show you the steps to crawl XKCD.
+
+First off, the `core` expects, both for `init` and `update`, a message compliant to the [schema definition for https strategies](https://github.com/neume-network/schema/blob/main/src/schema.mjs).
+
+the `init` function message should provide the starting state of the crawling. For XKCD, we would write something like:
+```
+{
+    write: null,
+    messages: [
+      {
+        type: "https",
+        commissioner: name,
+        version,
+        options: {
+          url: https://xkcd.com/1/info.0.json,
+          method: "GET",
+          header: null,
+          body: null
+        },
+        results: null,
+        error: null,
+      },
+    ],
+  };
+```
+
+* `write`: specifies the data to be written in string format. `null` during init, as there is no data
+* `type`: the protocol. must be `https` for this type of strategy
+* `commissioner`: ??
+* `version`: can be anything, it's currently not used
+* `options`:
+  * `url`: the URL pointing to the first chunk to crawl. First page on XKCD is `1`
+  * `method`: GET, POST
+  * `headers`: if any specific header should be sent to the source, it shall be specified here (for instance authorization headers)
+  * `body`: same
+* `results`: this field will be used by the core to pass the output of the crawling to the `update` function
+
+Once the `init` message has been returned, the `core` will fetch the first chunk of data and pass it to `update`. 
+
+The function, ideally, should take care of the following:
+
+* validate the outcome of the crawl
+* validate the data against a schema
+* prepare the data for storing. This step should make sure that any future consumer of the data will find all that's required to process it.
+* define if and what to crawl next
+
+The `core` provided message will contain the following data:
+```
+{
+    "error": string,
+    "results": object
+}
+```
+
+* error: null or the crawling error message
+* results: the direct output of the crawler
+
+The strategy will have to
```

Comment:
About the sections below suggesting to exit the entire crawl, I'd say this suggestion may not produce good outcomes. I'd prefer an inconsistent but complete crawl over the inverse.

So usually, in case of errors we log the message to improve debugging experience. And we try to deal with errors or validation failures in ways where we can make the best of the current crawl.

But certainly, there are cases where e.g. we could find that a continuing a crawl makes no sense anymore e.g. in case we expected a certain schema but then the data source is now returning another one.

@sirnicolaz does this help? Happy to assist with the formulation of the text, please let me know.

Reply:
Thanks for you comment. So are you saying that crawl should just log and continue upon error, unless it's a schema validation problem, for which case the crawl should halt?

Reply:
I changed the comment on the first part (mentioning that if possible, the crawling should go on). let me know if it's sufficient @TimDaub